ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitKudf.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.string;
+
+import com.google.common.base.Splitter;
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import java.util.Arrays;
+import java.util.List;
+
+@UdfDescription(name = SplitKudf.NAME, author = "Confluent",
+    description = "Splits a string into an array of substrings based on a delimiter.")
```

Comment:
you might want to add more docs here, i.e., what you put in the PR message:
```
If the delimiter is found at the beginning of the string, end of the string, or there are contiguous delimiters in the string, then empty strings are added to the array.
```

Reply:
Done

Reply:
can we (maybe you already did?!) do a little research on how the equivalent function in, say, MySQL/Presto/Oracle behaves in these cases ? Better to be consistent with them where possible than with Java libraries

Reply:
Ye,s I did. I looked at Presto, Teradata and Snowflake. All of the them are similar. Snowflake has more information about the beginning, end or continuous behavior. The others do not have such info, and I did not have a testbed to verify it, so I decided do what Snowflake does in this case.

Reply:
cool, thanks. That's one of the reasons i suggest MySQL btw - it's super easy to run in a docker container ;)  , in addition to being a familiar reference point for a lot of users

Reply:
Yeah, I do have a LXC container with Mysql installed too. But Mysql does not support such function. I just saw that Postgres supports it as SPLIT_PART(), which I could have used it for testing as well. Next time I will test Postgres too.